MailCore/Cache/MailboxManager/MailboxManager+Message.swift

```diff
@@ -49,7 +49,14 @@ public extension MailboxManager {
             throw CancellationError()
         }
 
-        let data = try await apiFetcher.attachment(attachment: attachment, progressObserver: progressObserver)
+        let attachmentCacheHelper = AttachmentCacheHelper()
+        let data: Data
+        if let cachedData = attachmentCacheHelper.getCache(resource: attachment.resource) {
+            data = cachedData
+        } else {
+            data = try await apiFetcher.attachment(attachment: attachment, progressObserver: progressObserver)
+            attachmentCacheHelper.storeCache(resource: attachment.resource, data: data)
+        }
```

Comment:
We can also move the realm write transaction in the else. We don't need to update it if it is already in cache.